### DISCUSSION TITLE
Reforged Eden

### DISCUSSION THREAD

**Discussion by SnakeSel** [SnakeSel/Empyrion-ru#3] [General]
В ходе перевода RE выявилась неприятная особенность. Описание некоторых предметов указано не ключами, а дано напрямую в файлах `Content/Configuration/*.ecf`
Приходится их редактировать.
Чтобы было проще их переводить при обновлении, я размещаю неболшие скрипты по замене текста в директории Util. 
Файл скрипта имеет имя соотвествующее файлу в Configuration который он правит.

**Comment by All4X4** [comment]
> В ходе перевода RE выявилась неприятная особенность. Описание некоторых предметов указано не ключами, а дано напрямую в файлах `Content/Configuration/*.ecf` Приходится их редактировать.

Лучше не надо. Без конфиг файлов работы много. Впрочем RE похоже уже тупиковый сценарий и особо его менять не будут. Они на RE2 сейчас сфокусировались.
 Я даже особо не припомню какие из item (с которыми взаимодействуют игроки) в игре не затронул бы лок.файл. Кроме Токенов.

 А при переводе ПДА и Диалогов, лучше дублировать англ.названия предметов миссии из лок файла. Кроме этого не переводить названия торг.компаний НПС, Планеты (заданий) и регионы/области фракций карты Галактики. И собственные Имена НПС (или дублировать англ) Типа Khalad. - этим грешат все машинные переводы (особенно когда часть переведёт, а часть нет/переведёт в одном месте, а в другом оставит не тронутым и тд).

 В любом случае решать тебе. Я тут "постольку-поскольку", некий вклад чернового перевода/правок внёс ещё в локу Адама (по сути там от Адама только машинный перевод) когда отлавливал битые теги. Теги вроде отловил, по крайней мере на которые жаловались игроки, а правки текста были побочным продуктом. Кстати лок файл там почему-то не полный, сравнил со своим архивом который отправлял Адаму.

**Comment by SnakeSel** [comment]
Я пока залез в то, что сразу бросается в глаза.
Описание схем как раз в ItemsConfig.ecf

Не переводить названия планет, расс, имена - согласен. Занес в глоссарий

**Comment by All4X4** [comment]
IMHO, зря полез в айтемы. В таких случаях пишется авторам сценария (сейчас это не актуально для РЕ 1), чтоб они присвоили ключ и вынесли его в лок файл. Тем более эта бронька не очень распространенная, и игроки уже привыкли юзать Ёлочку, так что не критично. Второй момент, мнительные не будут менять айтем файл, как и те кто юзает измененные конфиг файлы. Работы много и без этого.

К примеру, прочти ванильный перевод описание ПНВ, это тот самый случай, когда не особо заморачивались: подсветка шлема - нашлемный фонарь. И само описание надо сделать более понятное для новичков (а олды это не читают).

**Comment by All4X4** [comment]
Чтоб сократить заголовки (названия айтемов) нужно классифицировать их, даже если нет этого в оригинале. К0 (редко) К1...К5. Сильно сокращает "знакоместо" на русском, без лишних переносов слов, и остаётся понятным для игроков. Что-то типа:

Improved Heavy Armor  -   Улучшенная Тяжёлая Броня    -  Тяжёлая Броня К1
Advanced Heavy Armor   -  Продвинутая Тяжёлая Броня  -  Тяжёлая Броня К2

(и это еще не самые длинные названия, как правило вроде бы короткое название на англ, на русском получается очень длинное)

эти "продвинутые -улучшенные" не информативны и многабукф, параметры айтемов всё равно указываются в таблице внизу описания. В самом описании можно оставлять, но не в заголовках/названиях. И ещё, надо отслеживать по всему файлу все названия связанные с Improved-Advanced. Например Advanced, часто в тексте каша из слов: Продвинутая-Передовая-Расширенная-Усовершенствованная-Современный- Развитая и т.д.  **Названия у всех айтемов, например, со словом "Advanced" должно быть с единым "определением"**.

**Comment by All4X4** [comment]
Обратил внимание на глоссарий.

Augment - увеличение, Augmented - больше всего подходит "Дополненный" (слово короче). А под "модернизацию" попадают все типы улучшений, исключая стоковое (К0).

Еще обратил внимание и это, ИМХО, упущение игроделов. По НПС,  я пришел к своему варианту:

`ZiraxLaser,Laser Zirax,,,,,,,,Солдат (лазер),,,,,,,,`

Если переводить 
`ZiraxLaser,Laser Zirax,,,,,,,,Зиракс (лазер),,,,,,,,`

То в игре мы видим маркер "[Zirax] Зиракс (лазер)", потому что принадлежность автоматом подставляется. В моём варианте это 

[Zirax]Солдат (лазер)

![изображение](https://github.com/SnakeSel/Empyrion-ru/assets/164763695/c2b82ef5-b5db-4aa3-aa93-38a82002a1fe)
